app/javascript/tests/public_health/bulk_actions/CloseRecords.test.js

```diff
@@ -0,0 +1,94 @@
+  return wrapper;
+}
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('CloseRecords', () => {
+  it('Sets intial state after mount correctly', () => {
+    const wrapper = getWrapper([mockPatient1, mockPatient2, mockPatient6]);
+    expect(wrapper.state('apply_to_household')).toBeFalsy();
+    expect(wrapper.state('loading')).toBeFalsy();
+    expect(wrapper.state('monitoring')).toBeFalsy();
+    expect(wrapper.state('monitoring_reason')).toEqual('');
+    expect(wrapper.state('reasoning')).toEqual('');
+  });
+
+  it('Properly renders all main components', () => {
+    const wrapper = getWrapper([mockPatient1, mockPatient2, mockPatient6]);
+    expect(wrapper.find('p').find('span').at(0).text()).toEqual('You are about to change the Monitoring Status of the selected records from "Actively Monitoring" to "Not Monitoring".');
+    expect(wrapper.find('p').find('span').at(1).text()).toEqual(' These records will be moved to the closed line list and the reason for closure will be blank.');
+    expect(wrapper.find('FormGroup').length).toEqual(3);
+    expect(wrapper.find('FormGroup').at(0).find('FormLabel').text()).toEqual('Please select reason for status change:');
+    expect(wrapper.find('#monitoring_reason').exists()).toBeTruthy();
+    const monitoringReasonOptions = [''].concat(mockMonitoringReasons);
+    wrapper
+      .find('#monitoring_reason')
+      .find('option')
+      .forEach((option, index) => {
+        expect(option.text()).toEqual(monitoringReasonOptions[Number(index)]);
+      });
+    expect(wrapper.find('FormGroup').at(1).find('FormLabel').length).toEqual(2);
+    expect(wrapper.find('FormGroup').at(1).find('FormLabel').at(0).text()).toEqual('Please include any additional details:');
+    expect(wrapper.find('FormGroup').at(1).find('FormLabel').at(1).text()).toContain('characters remaining');
+    expect(wrapper.find('#apply_to_household').exists()).toBeTruthy();
+    expect(wrapper.find('ModalFooter').find('Button').at(0).text()).toEqual('Cancel');
+    expect(wrapper.find('ModalFooter').find('Button').at(1).text()).toEqual('Submit');
+  });
+
+  it('Changing monitoring reason dropdown updates state', () => {
+    const wrapper = getWrapper([mockPatient1, mockPatient2, mockPatient6]);
+
+    // initial modal state with monitoring reason empty
+    expect(wrapper.state('monitoring_reason')).toEqual('');
+    expect(wrapper.find('p').find('span').at(1).exists()).toBeTruthy();
+
+    // test changing to each enabled monitoring option
+    mockMonitoringReasons.forEach(value => {
+      wrapper.find('#monitoring_reason').simulate('change', { target: { id: 'monitoring_reason', value: value } });
+      expect(wrapper.state('monitoring_reason')).toEqual(value);
+      expect(wrapper.state('monitoring')).toEqual(false);
+      expect(wrapper.find('p').find('span').at(1).exists()).toBeFalsy();
+    });
+  });
+
```

Comment:
can we add a test for changing the additional details text input and the character count updating?